webprotege-gwt-ui-client/src/main/java/edu/stanford/bmir/protege/web/client/logicaldefinition/LogicalDefinitionPortletViewImpl.java

```diff
@@ -0,0 +1,288 @@
+import edu.stanford.bmir.protege.web.shared.frame.PropertyClassValue;
+import edu.stanford.bmir.protege.web.shared.icd.AncestorClassHierarchy;
+import edu.stanford.bmir.protege.web.shared.icd.GetClassAncestorsAction;
+import edu.stanford.bmir.protege.web.shared.logicaldefinition.GetEntityLogicalDefinitionAction;
+import edu.stanford.bmir.protege.web.shared.logicaldefinition.GetEntityLogicalDefinitionResult;
+import edu.stanford.bmir.protege.web.shared.logicaldefinition.LogicalDefinition;
+import edu.stanford.bmir.protege.web.shared.logicaldefinition.SaveLogicalDefinitionAction;
+import edu.stanford.bmir.protege.web.shared.postcoordination.*;
+import edu.stanford.bmir.protege.web.shared.project.ProjectId;
+import org.semanticweb.owlapi.model.IRI;
+import org.semanticweb.owlapi.model.OWLClass;
+import org.semanticweb.owlapi.model.OWLEntity;
+import org.semanticweb.owlapi.model.OWLObjectProperty;
+import uk.ac.manchester.cs.owl.owlapi.OWLObjectPropertyImpl;
+
+import javax.inject.Inject;
+import java.util.*;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class LogicalDefinitionPortletViewImpl extends Composite implements LogicalDefinitionPortletView {
+
+    Logger logger = java.util.logging.Logger.getLogger("LogicalDefinitionPortletViewImpl");
+
+    @UiField
+    HTMLPanel paneContainer;
+
+    @UiField
+    HTMLPanel superClassContainer;
+
+    @UiField
+    HTMLPanel necessaryConditionsContainer;
+    @UiField
+    ListBox ancestorDropdown;
+
+    @UiField Button editValuesButton;
+    @UiField Button cancelButton;
+
+    @UiField Button saveValuesButton;
+
+    private static final LogicalDefinitionPortletViewImpl.LogicalDefinitionPortletViewImplUiBinder ourUiBinder = GWT.create(LogicalDefinitionPortletViewImpl.LogicalDefinitionPortletViewImplUiBinder.class);
+
+    private final DispatchServiceManager dispatchServiceManager;
+
+    private ProjectId projectId;
+
+    private List<OWLEntityData> ancestorsList = new ArrayList<>();
+
+    private List<PostCoordinationTableAxisLabel> labels;
+
+    private WhoficCustomScalesValues superclassScalesValue;
+
+    private WhoficEntityPostCoordinationSpecification superclassSpecification;
+
+    private List<PostcoordinationAxisToGenericScale> axisToGenericScales = new ArrayList<>();
+    private LogicalDefinitionResourceBundle.LogicalDefinitionCss style;
+
+    private boolean readOnly = true;
+
+
+    private final LogicalDefinitionTable superClassTable = new LogicalDefinitionTable(new LogicalDefinitionTableConfig("Logical Definition Axis",
+            "Value",
+            this::initializeTable));
+
+
+    private final LogicalDefinitionTable necessaryConditionsTable = new LogicalDefinitionTable(new LogicalDefinitionTableConfig("Necessary Axis",
+            "Value",
+            this::initializeTable));
+
+    @Inject
+    public LogicalDefinitionPortletViewImpl(DispatchServiceManager dispatchServiceManager) {
```

Comment:
[sugestion]: from what i've seen from other view components in the application most of the actions that are made in a view should be handled by functional interfaces and the actual action should be defined in the presenter.
Normally i don't think you want to make dispatch calls and other major actions directly from the view. The view should be more or less just for showing the model, the presenter should do operations on the model and dictate what the view presents.
I think it's going to be a headache to change any of this now, but we should keep it in mind if we need to implement anything else.